File: src/practice/luogu/p1601/Main.java
// https://www.luogu.com.cn/problem/P1601
package practice.luogu.p1601;

import java.math.BigInteger;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String a = sc.nextLine();
        String b = sc.nextLine();
        System.out.println(new BigInteger(a).add(new BigInteger(b)));
        sc.close();
    }
}

File: src/practice/luogu/p1601/Main.cpp
// https://www.luogu.com.cn/problem/P1601
#include <bits/stdc++.h>
using namespace std;
#define max_n 520
int a[max_n], b[max_n], c[max_n];
int main() {
  string A, B;
  cin >> A >> B;
  int len = max(A.length(), B.length());
  for (int i = A.length() - 1, j = 1; i >= 0; i--, j++) {
    a[j] = A[i] - '0';
  }
  for (int i = B.length() - 1, j = 1; i >= 0; i--, j++) {
    b[j] = B[i] - '0';
  }
  for (int i = 1; i <= len; i++) {
    c[i] += a[i] + b[i];
    c[i + 1] += c[i] / 10;
    c[i] = c[i] % 10;
  }
  if (c[len + 1]) {
    len++;
  }
  for (int i = len; i >= 1; i--) {
    cout << c[i];
  }
  cout << endl;
  return 0;
}